src/parquet/encodings/encodings.h

```diff
@@ -62,12 +68,13 @@ class Decoder {
   const parquet::Encoding::type encoding() const { return encoding_; }
 
  protected:
-  Decoder(const parquet::Type::type& type, const parquet::Encoding::type& encoding)
-    : type_(type), encoding_(encoding), num_values_(0) {}
+  Decoder(const parquet::Type::type& type, const parquet::Encoding::type& encoding,int fixed_length)
+    : type_(type), encoding_(encoding), num_values_(0), fixed_length_size(fixed_length) {}
 
   const parquet::Type::type type_;
   const parquet::Encoding::type encoding_;
   int num_values_;
+  int fixed_length_size;
```

Comment:
Let's revisit after #22 is reviewed and merged, but having `fixed_length_size` "leak" through to affect all of the decoders is evidence that the Decoder API needs to be changed. I think that the Decoder constructors should accept `parquet::SchemaElement*` so that type-specific metadata can be accessed in the appropriate place and nowhere else.